src/OrbitGl/App.cpp

```diff
@@ -1347,49 +1348,44 @@ void OrbitApp::EnableFrameTracksFromHashes(const ModuleData* module,
 }
 
 void OrbitApp::LoadPreset(const std::shared_ptr<PresetFile>& preset_file) {
-  std::vector<ModuleData*> modules_to_load;
-  std::vector<std::string> module_paths_not_found;  // file path of module
-  for (const auto& [module_path, preset_module] : preset_file->preset_info().path_to_module()) {
-    ModuleData* module_data = module_manager_->GetMutableModuleByPath(module_path);
+  std::vector<orbit_base::Future<std::string>> load_module_results{};
+  load_module_results.reserve(preset_file->preset_info().path_to_module().size());
 
-    if (module_data == nullptr) {
-      module_paths_not_found.push_back(module_path);
-      continue;
-    }
-    if (module_data->is_loaded()) {
-      std::vector<uint64_t> function_hashes{preset_module.function_hashes().begin(),
-                                            preset_module.function_hashes().end()};
-      SelectFunctionsFromHashes(module_data, function_hashes);
-      std::vector<uint64_t> frame_track_hashes{preset_module.frame_track_function_hashes().begin(),
-                                               preset_module.frame_track_function_hashes().end()};
-      EnableFrameTracksFromHashes(module_data, frame_track_hashes);
-      continue;
-    }
-    modules_to_load.push_back(module_data);
-  }
-  if (!module_paths_not_found.empty()) {
-    // Note that unloadable presets are disabled.
-    SendWarningToUi("Preset only partially loaded",
-                    absl::StrFormat("The following modules are not loaded:\n\"%s\"",
-                                    absl::StrJoin(module_paths_not_found, "\"\n\"")));
-  }
-  if (!modules_to_load.empty()) {
-    absl::flat_hash_map<std::string, std::vector<uint64_t>> function_hashes_to_hook_map;
-    absl::flat_hash_map<std::string, std::vector<uint64_t>> frame_track_function_hashes_map;
-    for (const auto& [module_path, preset_module] : preset_file->preset_info().path_to_module()) {
-      function_hashes_to_hook_map[module_path] = std::vector<uint64_t>{};
-      for (const uint64_t function_hash : preset_module.function_hashes()) {
-        function_hashes_to_hook_map.at(module_path).push_back(function_hash);
-      }
-      frame_track_function_hashes_map[module_path] = std::vector<uint64_t>{};
-      for (const uint64_t function_hash : preset_module.frame_track_function_hashes()) {
-        frame_track_function_hashes_map.at(module_path).push_back(function_hash);
-      }
+  // First we try to load all preset modules in parallel
+  for (const auto& [module_path, preset_module] : preset_file->preset_info().path_to_module()) {
+    auto load_preset_result = LoadPresetModule(module_path, preset_module);
+
+    orbit_base::ImmediateExecutor immediate_executor{};
+    auto future = load_preset_result.Then(
+        &immediate_executor,
+        [module_path = module_path](const ErrorMessageOr<void>& result) -> std::string {
```

Comment:
What is this `[module_path = module_path]` and how does it differ to `[module_path]`?

Reply:
Is it because `const auto& [module_path, preset_module]` just creates an alias, or something in this direction?

Reply:
Yes, you're on the right track. You can't capture a binding.